int16 number
---
int16 size